src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/OAuth2/OAuthClient.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Core.Http;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Helpers;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.OAuth2;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Platform;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Core.OAuth2
+{
+    internal class OAuthClient
```

Comment:
>OAuthClient [](start = 19, length = 11)

is this a copy of what is in MSAL? is it a copy of the AdalHttpClient? or does it take extra responsibilities?

Reply:
this is closer to a copy of AdalHttpClient, but with a name similar to what we have in MSAL (but can change the name). 

---
In reply to: [267507862](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1530#discussion_r267507862) [](ancestors = 267507862)